pkg/polaris/graphql/graphql.go

```diff
@@ -149,17 +151,40 @@ const requestRetryAttempts = 10
 // Request posts the specified GraphQL query/mutation with the given variables
 // to the Polaris platform. Returns the response JSON text as is. If the request
 // fails due to temporary error, it will be retried automatically.
-func (c *Client) Request(ctx context.Context, query string, variables interface{}) ([]byte, error) {
+func (c *Client) Request(ctx context.Context, query string, variables any) ([]byte, error) {
 	c.log.Print(log.Trace)
 
-	// Log variables before calling the query/mutation.
-	buf, err := json.Marshal(variables)
+	// Log query and variables.
+	buf, err := json.MarshalIndent(secret.Redact(variables), "", "    ")
 	if err != nil {
-		buf = []byte(fmt.Sprintf("marshaling of variables failed: %s", err))
+		return nil, fmt.Errorf("failed to marshal query variables for logging: %s", err)
 	}
-	c.log.Printf(log.Debug, "%s params: %s", QueryName(query), string(buf))
+	c.log.Printf(log.Debug, "%s request:\n%s\n%s", QueryName(query), query, string(buf))
 
-	return c.RequestWithoutLogging(ctx, query, variables)
+	// Do request.
+	res, err := c.RequestWithoutLogging(ctx, query, variables)
+	if err != nil {
+		return nil, err
+	}
+
+	// Log the data field of the response.
```

Comment:
The responses might include secrets too, could be worth investigation how to add redaction for those as well.

Reply:
Yes! Good idea, I'll create a ticket for it.